### PR TITLE
Add pagination

### PR DESCRIPTION
**Pull request by RomainGuarinoni**
### Offset pagination
We added pagination system for these routes : 

- GET /musicians
- GET / events
- GET / groups

The pagination system is based on the **offset pagination** with  a **start** and a **limit** query options. _By default, start is set to 0 and limit is set to 20._

### _links

The `_links` object return by the API give contains the URLs to query the self, first , next  and previous page.For example, if the user want to have the next page, the frontend have to fetch the `_links.next` url. The `_links` object may not contains `previous` if there is no items before the current query, and may not contains `next` if there is no items after the current query.

See these docs for more info : [the current query](https://developer.atlassian.com/server/confluence/pagination-in-the-rest-api/)

### Additionnal information

The response objects also give this informations : 

- **size** : The size of the results array
- **limit** : The limit of the query
- **start** : The start of the query
- **total** : The numbers of entries in total in the db for this request

Here is the result for this request : `http://localhost:8000/groups?start=2&limit=2`

```json
{
{
  "results": [
    {
      "id": "71cb6241-a707-4cde-ad92-8d2c969d7e79",
      "name": "All that Remains",
      "description": "metal core à l'ancienne",
      "location": "Lille",
      "genres": [
        {
          "id": "6df6f90e-c02c-48b8-9a6b-fee1ab47e3b3",
          "name": "metal"
        }
      ],
      "members": [
        {
          "membership": "member",
          "musician": {
            "id": "05984b2d-0907-4e50-9c5e-f08bbdab02a1",
            "email": "romain.guar91@gmail.com",
            "description": "Je suis romain guarinoni. Un vrai boute-en-train ! Je code et je fais de la musique. Dans ces 2 domaines je suis très problement bien meilleur que toi désolé fréro le sang.",
            "givenName": "Romain",
            "familyName": "Guarinoni",
            "phone": null,
            "facebookUrl": null,
            "twitterUrl": null,
            "instagramUrl": null,
            "promotion": "M1",
            "location": "Douai",
            "password": "$2b$10$oSkbT5bDabLIuJ5ikcdKT.11kquR8q4MU7sbgo1tFWU67AgnOcppu",
            "isLookingForGroups": false
          },
          "instruments": [
            {
              "id": "86504bb7-7e66-43d7-8bbc-f37dd7cd8fc7",
              "name": "guitare"
            }
          ]
        }
      ]
    },
    {
      "id": "70ef1ce4-1a99-4d74-8b0a-0731da4f832f",
      "name": "jazz band",
      "description": "du jazz",
      "location": "Douai",
      "genres": [
        {
          "id": "18dcafa1-a179-4e1b-b626-a46f07646efc",
          "name": "jazz"
        }
      ],
      "members": [
        {
          "membership": "member",
          "musician": {
            "id": "d267c7fb-bee8-4e6c-a0a8-407c0fb1ecd6",
            "email": "dorian.viala@gmail.com",
            "description": null,
            "givenName": "Dorian",
            "familyName": "Viala",
            "phone": null,
            "facebookUrl": null,
            "twitterUrl": null,
            "instagramUrl": null,
            "promotion": "M1",
            "location": "Douai",
            "password": "$2b$10$ebn7aGHs7nvFmyg0PIeaU.z0NegP/ikxX9rt7na4nZuJh9NA60BIq",
            "isLookingForGroups": false
          },
          "instruments": [
            {
              "id": "86504bb7-7e66-43d7-8bbc-f37dd7cd8fc7",
              "name": "guitare"
            }
          ]
        }
      ]
    }
  ],
  "size": 2,
  "limit": 2,
  "start": 2,
  "total": 5,
  "_links": {
    "self": "http://localhost:8000/groups",
    "first": "http://localhost:8000/groups?start=0&limit=2",
    "previous": "http://localhost:8000/groups?start=0&limit=2",
    "next": "http://localhost:8000/groups?start=4&limit=2"
  }
}
```

### CODE REVIEW

File: api/routes/events/events.ts [hunk 1]
```diff
@@ -103,9 +117,42 @@ router.get(
           }
         },
         relations: ['genres', 'groups', 'admins', 'groups.genres'],
+        skip: start,
```
Comment (alexandrelam) [inline api/routes/events/events.ts:120]: shouldn't skip be like `limit * skip` ?

Comment (alexandrelam) [inline api/routes/events/events.ts:120]: shouldn't skip be like `limit * skip` ?

File: api/routes/events/events.ts [hunk 2]
```diff
@@ -41,6 +41,20 @@ router.get(
     res: core.Response<getResponsesBody<GetEvents>, {}, getHTTPCode<GetEvents>>,
   ) => {
     try {
+      // Pagination
+      const start =
```
Comment (alexandrelam) [inline api/routes/events/events.ts:45]: can you name that page ? or skip.

**Review by alexandrelam — CHANGES_REQUESTED** [review]

File: api/routes/events/events.ts (hunk 1 above)
Comment (RomainGuarinoni) [reply api/routes/events/events.ts:120]: Same

File: api/routes/events/events.ts (hunk 2 above)
Comment (RomainGuarinoni) [reply api/routes/events/events.ts:45]: As i said, it's not a page but an start ( offset)

File: api/routes/events/events.ts (hunk 1 above)
Comment (alexandrelam) [reply api/routes/events/events.ts:120]: i like page better, it's easier to understand

**Pull request merged**